test/test_data_collector.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import errno
+import os
+from pathlib import Path
+import time
+
+from ros_cross_compile.data_collector import DataCollector
+from ros_cross_compile.data_collector import DataTimer
+from ros_cross_compile.data_collector import Datum
+
+
+def test_construction():
+    test_datum = Datum('test_stat', 3, 'tests', '3 PM PST')
+    test_collector = DataCollector()
+    assert test_datum
+    assert test_collector
+
+
+def test_data_collection():
+    test_collector = DataCollector()
+    test_collector.set_write_path(Path.cwd())
```

Comment:
not going to work as `cwd` may not be writeable.

Reply:
(extreme example):

```sh
$ cd /
$ python /tmp/path/to/cross_compile/test.py```
```

Here cwd is  "/", "/" can't be written in.

Reply:
One option is to use a temporary directory: https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir

Another option is to avoid having write API use paths directly and allow buffers instead.